hash/hashing.go

```diff
@@ -0,0 +1,15 @@
+package hash
```

Comment:
This might be confusing to the reader if they expect the golang `hash` package. But let's tackle the general package structuring in followup PRs _before_ we do the first _major_ release. 
related to https://github.com/lazyledger/optimint/pull/79#discussion_r647321719